src/selectors/entities/channels.js

```diff
@@ -923,3 +923,26 @@ export const getSortedDirectChannelWithUnreadsIds = createIdsSelector(
     filterChannels,
 );
 
+// Filters post IDs by the given condition.
+// The condition function receives as parameters the associated channel object and the post object.
+export const filterPostIds = (condition) => {
+    if (typeof condition !== 'function') {
+        throw new TypeError(`${condition} is not a function`);
+    }
+    return createSelector(
+        getAllChannels,
+        getAllPosts,
+        (state, postIds) => postIds,
+        (channels, posts, postIds) => {
+            return postIds.filter((postId) => { // eslint-disable-line
+                const post = posts[postId];
+                if (post) {
```

Comment:
Filter expect a function that returns a boolean (it is working because the id is like a boolean in this case).

I think you can replace it with something like
```javascript
const post = posts[postId];
return post && condition(channels[post.channel_id], post)
```

Reply:
And besides, you don't need to remove the es-lint rule this way.

Reply:
Seems to me that filter just handles no return correctly, try this:

```js
['hello', 1, null, {}, 'b'].filter(item => {
	if (typeof item === 'string') {
		return item;
    }
});
```
// ["hello", "b"]

Reply:
Yes, filter handle no-return properly because no-returns is equal to undefined, and undefined is false. and filter expect a function that returns truish o falsish values. You are returning the item, which is at least, not what is expected by filter. Works, but is not the expected usage.

Reply:
Yeah it's an implicit return... I guess being explicit is probably better.